src/devtools/packages/devtools-reps/object-inspector/components/ObjectInspector.tsx

```diff
@@ -98,23 +112,24 @@ class OI extends PureComponent<ObjectInspectorProps> {
     } else {
       this.expandedPaths.delete(item.path);
     }
-    this.forceUpdate();
 
     if (!expand) {
+      this.forceUpdate();
       return;
     }
 
     if (item.type === "getter" && item.valueItem) {
       item = item.valueItem;
     }
-    if (item.type === "value" && item.needsToLoadChildren()) {
+    if ("childrenLoaded" in item && !item.childrenLoaded) {
       try {
         await item.loadChildren();
       } catch {
         this.expandedPaths.delete(item.path);
       }
-      this.forceUpdate();
     }
```

Comment:
This comment higher level than the PR, but as I'm looking at this code now I'm noticing that the way it uses instance properties may cause problems when we eventually migrate to `createRoot` and start using React's concurrent features. Maybe we could chat some time about why we're putting things on the instance rather than in component state. 😁

Reply:
I don't know the reason, I kept it from legacy code. Given how hard it can be in my experience to reason about ObjectInspector's update strategy, I would very much welcome a thorough refactoring of this code.